tests/test_vxlan_decap.py

```diff
@@ -0,0 +1,149 @@
+    @mg_facts: Minigraph facts
+    """
+    loopback_ip = None
+    for intf in mg_facts["minigraph_lo_interfaces"]:
+        if IPAddress(intf["addr"]).version == 4:
+            loopback_ip = intf["addr"]
+            break
+    if not loopback_ip:
+        pytest.fail("ipv4 lo interface not found")
+
+    # Generate vxlan tunnel config json file on DUT
+    vxlan_tunnel_cfg = {
+        "VXLAN_TUNNEL": {
+            "tunnelVxlan": {
+                "src_ip": loopback_ip,
+                "dst_ip": VTEP2_IP
+            }
+        }
+    }
+    duthost.copy(content=json.dumps(vxlan_tunnel_cfg, indent=2), dest="/tmp/vxlan_db.tunnel.json")
+
+    # Generate vxlan maps config json file on DUT
+    vxlan_maps_cfg = {
+        "VXLAN_TUNNEL_MAP": {}
+    }
+    for vlan in mg_facts["minigraph_vlans"]:
+        vxlan_maps_cfg["VXLAN_TUNNEL_MAP"]["tunnelVxlan|map%s" % vlan] = {
+            "vni": int(vlan.replace("Vlan", "")) + VNI_BASE,
+            "vlan": vlan
+        }
+    duthost.copy(content=json.dumps(vxlan_maps_cfg, indent=2), dest="/tmp/vxlan_db.maps.json")
+
+
+@pytest.fixture(scope="module")
+def setup(duthost, ptfhost):
+
+    logger.info("Gather some facts")
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)["ansible_facts"]
+    dut_facts = duthost.setup(gather_subset="!all,!any,network", filter="ansible_Ethernet*")["ansible_facts"]
+    ptf_facts = ptfhost.setup(gather_subset="!all,!any,network")["ansible_facts"]
+
+    logger.info("Prepare PTF")
+    prepare_ptf(ptfhost, mg_facts, dut_facts)
+
+    logger.info("Generate VxLAN config files")
+    generate_vxlan_config_files(duthost, mg_facts)
+
+    setup_info = {
+        "mg_facts": mg_facts
+    }
+
+    yield setup_info
+
+    logger.info("Stop arp_responder on PTF")
+    ptfhost.shell("supervisorctl stop arp_responder")
+
+    logger.info("Always try to remove any possible VxLAN tunnel and map configuration")
+    for vlan in mg_facts["minigraph_vlans"]:
+            duthost.shell('docker exec -i database redis-cli -n 4 -c DEL "VXLAN_TUNNEL_MAP|tunnelVxlan|map%s"' % vlan)
+    duthost.shell('docker exec -i database redis-cli -n 4 -c DEL "VXLAN_TUNNEL|tunnelVxlan"')
+
+
+@pytest.fixture(params=["NoVxLAN", "Enabled", "Removed"])
+def vxlan_status(setup, request, duthost):
+    if request.param == "Enabled":
+        duthost.shell("sonic-cfggen -j /tmp/vxlan_db.tunnel.json --write-to-db")
+        duthost.shell("sonic-cfggen -j /tmp/vxlan_db.maps.json --write-to-db")
+        return True, request.param
+    elif request.param == "Removed":
+        for vlan in setup["mg_facts"]["minigraph_vlans"]:
+            duthost.shell('docker exec -i database redis-cli -n 4 -c DEL "VXLAN_TUNNEL_MAP|tunnelVxlan|map%s"' % vlan)
+        duthost.shell('docker exec -i database redis-cli -n 4 -c DEL "VXLAN_TUNNEL|tunnelVxlan"')
+        return False, request.param
+    else:
+        return False, request.param
+
+
+def test_vxlan_decap(setup, vxlan_status, duthost, ptfhost):
```

Comment:
Where can I find the vxlan_status values?

Reply:
vxlan_status is python fixture.

Reply:
Yes, vxlan_status is defined as a pytest fixture. It will be executed 3 times. For each execution, it is supplied with a param value defined in list ["NoVxLAN", "Enabled", "Removed"] and returns a different tuple based on the param value. 
The returned values are sent to the test_vxlan_decap case. This means that the test_vxlan_decap case will also be executed 3 times. Each time the test case is supplied with different return value from vxlan_status.